### DISCUSSION TITLE
Deploy a express app with typescript to vercel

### DISCUSSION THREAD

**Discussion by Jakob5358** [vercel/community#411] [Help] [answered]
I've been trying for a while, but I can't quite get it to work. Either it returns a 404 or it just returns the index.js code when visiting the route.

I'm using typescript, meaning my code is in an src folder. When the build command is ran, it compiles the code into js in the /dist folder (I can change the output folder name if required).
How do I configure it, can I do it without vercel.json, if not, what should I put in there?

**Comment by swarnava** [comment, marked as answer]
Hi @Jakob5358,

Vercel is a Serverless and Edge-first platform and does not support running servers. Vercel is, however, committed to dynamic frontends coupled with [Serverless Functions](https://vercel.com/docs/concepts/functions/serverless-functions) and [Edge Functions](https://vercel.com/docs/concepts/functions/edge-functions).

You can find more information about why running servers is not possible using the Vercel platform in [this article](https://vercel.com/knowledge/npm-run-start-not-working).

**Reply by Jakob5358** [reply]
Oh, I was looking at this https://vercel.com/guides/using-express-with-vercel 
If you scroll down a bit there's a section about 'standalone express'. Is that deprecated or am I totally confused?

**Reply by Markos-Th09** [reply]
It isn't deprecated you can use it, though it isn't the best in in a serverless environment

**Reply by Jakob5358** [reply]
> It isn't deprecated you can use it, though it isn't the best in in a serverless environment

Ah ok thanks!
